File: iOS/CMS_SDK/Composr Mobile SDK/CMS_Utilities/CMS_Timestamps.m
//
//  CMS_Timestamps.m
//  Composr Mobile SDK
//
//  Created by Aaswini on 06/08/14.
//  Copyright (c) 2014 Aaswini. All rights reserved.
//

#import "CMS_Timestamps.h"

@implementation CMS_Timestamps

/**
 *  Turns a unix timestamp into a written date&time, in the phone's timezone
 *
 *  @param timestamp   unix timestamp to be converted
 *  @param includeTime should time be considered from the timestamp ? YES - consider time, NO - discard time
 *
 *  @return String representation of the supplied timestamp
 */
+ (NSString *)get_timezoned_date_time:(int)timestamp{
    NSDate *localDate = [NSDate dateWithTimeIntervalSince1970:timestamp];
    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc]init];
    
    [dateFormatter setDateFormat:@"MMM dd, yyyy - HH:mm:ss"];
    return [dateFormatter stringFromDate:localDate];
}

/**
 *  Turns a unix timestamp into a written date, in the phone's timezone
 *
 *  @param timestamp   unix timestamp to be converted
 *  @param includeTime should time be considered from the timestamp ? YES - consider time, NO - discard time
 *
 *  @return String representation of the supplied timestamp
 */
+ (NSString *)get_timezoned_date:(int)timestamp{
    NSDate *localDate = [NSDate dateWithTimeIntervalSince1970:timestamp];
    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc]init];
    
    [dateFormatter setDateFormat:@"MMM dd, yyyy"];
    return [dateFormatter stringFromDate:localDate];
}

/**
 *  Returns unix timestamp
 *
 *  @return returns unix timestamp
 */
+ (int)time{
    return [[NSDate date] timeIntervalSince1970];
}

@end

File: android/src/com/ocproducts/composr/sdk/utilities/CMS_Timestamps.java
package com.ocproducts.composr.sdk.utilities;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

/*

CMS_Timestamps

get_timezoned_date_time(int timestamp) - turns a unix timestamp into a written date&time, in the phone's timezone
get_timezoned_date(int timestamp) - turns a unix timestamp into a written date, in the phone's timezone
int time() - returns unix timestamp

*/

public class CMS_Timestamps {

	public static String get_timezoned_date_time(long timestamp) {
		SimpleDateFormat df;
	    
    	df = new SimpleDateFormat("MMM dd, yyyy - HH:mm:ss", Locale.getDefault());
	    
	    return df.format(new Date(timestamp));
	}
	
	public static String get_timezoned_date(long timestamp) {
		SimpleDateFormat df;
	    
    	df = new SimpleDateFormat("MMM dd, yyyy", Locale.getDefault());
	    
	    return df.format(new Date(timestamp));
	}
	
	public static long time() {
		return new Date().getTime();
	}
}
